packages/db/src/search/StatsSearch.ts

```diff
@@ -0,0 +1,298 @@
+import { Knex } from 'knex';
+import omit from 'lodash/omit';
+import { getKnexClient } from '../connection';
+import { TableNames } from '../tables';
+import { DbQueryParameters } from '../types/search';
+import { BaseSearch, typeToTable } from './BaseSearch';
+
+type TotalSummary = {
+  count_errors: number,
+  count_collections: number,
+  count_granules: number,
+  avg_processing_time: number,
+};
+
+type Aggregate = {
+  count: string,
+  status?: string,
+  error?: string,
+  name?: string,
+};
+
+type Summary = {
+  dateFrom: string,
+  dateTo: string,
+  value: number,
+  aggregation: string,
+  unit: string,
+};
+
+type SummaryResult = {
+  errors: Summary,
+  granules: Summary,
+  collections: Summary,
+  processingTime: Summary,
+};
+
+type Meta = {
+  name: string,
+  count: number,
+  field: string,
+};
+
+type AggregateRes = {
+  key: string,
+  count: number,
+};
+
+type ApiAggregateResult = {
+  meta: Meta,
+  count: AggregateRes[]
+};
+
```

Comment:
Can you just create a object?
const infixMapping = {
  granules: 'granule_id',

Reply:
But you can do infixMapping[get(typeToTable[this.type]], make it simple.